test/e2e/sched/sched_test.go

```diff
@@ -264,10 +272,300 @@ var _ = Describe("[Scheduler] CR configuration management", func() {
+			Expect(err).ToNot(HaveOccurred())
+			initialPodList, err := podlist.With(e2eclient.Client).ByDeployment(context.TODO(), *deployment)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(initialPodList).To(HaveLen(1), "cannot find correct amount of pods for DP %s/%s", deployment.Namespace, deployment.Name)
+			uid := initialPodList[0].UID
+
+			newValue := nropv1.CacheResyncDetectionAggressive
+			expectedCMValue := "All"
+			if initialSchedObj.Spec.CacheResyncDetection != nil && *initialSchedObj.Spec.CacheResyncDetection == newValue {
+				newValue = nropv1.CacheResyncDetectionRelaxed
+				expectedCMValue = "OnlyExclusiveResources"
+			}
+
+			By(fmt.Sprintf("modifying the NUMAResourcesScheduler spec.CacheResyncDetection field to %q", newValue))
+			nroSchedObj := &nropv1.NUMAResourcesScheduler{}
+			Eventually(func(g Gomega) {
+				//updates must be done on object.Spec and active values should be fetched from object.Status
+				err := e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(initialSchedObj), nroSchedObj)
+				g.Expect(err).ToNot(HaveOccurred())
+
+				nroSchedObj.Spec.CacheResyncDetection = &newValue
+				err = e2eclient.Client.Update(context.TODO(), nroSchedObj)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+			By("verify scheduler is available")
+			nroSchedObj = nrosched.CheckNROSchedulerAvailable(context.TODO(), e2eclient.Client, nroSchedObj.Name)
+			Expect(nroSchedObj).ToNot(BeNil())
+
+			By("verify scheduler CacheResyncDetection mode is updated")
+			var cm corev1.ConfigMap
+			cmKey := client.ObjectKey{
+				Name:      schedulerConfigMapName,
+				Namespace: deployment.Namespace,
+			}
+			Expect(e2eclient.Client.Get(context.TODO(), cmKey, &cm)).Should(Succeed())
+			data, ok := cm.Data[schedstate.SchedulerConfigFileName]
+			Expect(data).ToNot(BeEmpty(), "no data found under %s/%s", cm.Namespace, cm.Name)
+			Expect(ok).To(BeTrue(), "no data found under %s/%s", cm.Namespace, cm.Name)
+
+			schedParams, err := manifests.DecodeSchedulerProfilesFromData([]byte(data))
+			Expect(err).ToNot(HaveOccurred())
+			schedCfg := manifests.FindSchedulerProfileByName(schedParams, nroSchedObj.Status.SchedulerName)
+			Expect(schedCfg).ToNot(BeNil(), "cannot find profile config for profile %q", nroSchedObj.Status.SchedulerName)
+			Expect(schedCfg.Cache).ToNot(BeNil(), "missing cache configuration")
+			Expect(schedCfg.Cache.ForeignPodsDetectMode).ToNot(BeNil(), "missing cache resync configuration")
+			Expect(*schedCfg.Cache.ForeignPodsDetectMode).To(Equal(expectedCMValue))
+
+			By("verify the pod was restarted")
+			klog.Info("checking the old pod is removed")
+			err = wait.With(e2eclient.Client).Timeout(3*time.Minute).ForPodDeleted(context.TODO(), initialPodList[0].Namespace, initialPodList[0].Name)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
this is correct but not sufficient. We test:
- the operator created the expected config succesfully (good)
- the scheduler pod(s) was (were) restarted (good)

if both the above are correct, there's high confidence the config was picked up. But this is not sufficient.
We should add facilities in the scheduler (or our d/s fork) to reflect the config used in a easy to consume way. (no log parsing!)